test/fuzz/yaml/yaml.go

```diff
@@ -20,10 +20,51 @@ limitations under the License.
 package yaml
 
 import (
+	"bytes"
+
 	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	sigyaml "sigs.k8s.io/yaml"
 )
 
+// FuzzDuration is a fuzz target for unmarshaling Duration defined in "k8s.io/apimachinery/pkg/apis/meta/v1".
+// This target also checks that the unmarshaled result can be marshaled back to the input.
+func FuzzDuration(b []byte) int {
+	var unmarshalResult struct {
+		D metav1.Duration `json:"d"`
+	}
+	if err := yaml.Unmarshal(b, &unmarshalResult); err != nil {
+		return 0
+	}
+	marshalResult, err := yaml.Marshal(&unmarshalResult)
+	if err != nil {
+		panic(err)
+	}
+	if !bytes.Equal(marshalResult, b) {
+		panic("marshalResult != input")
+	}
+	return 1
+}
+
+// FuzzMicroTime is a fuzz target for unmarshaling MicroTime defined in "k8s.io/apimachinery/pkg/apis/meta/v1".
+// This target also checks that the unmarshaled result can be marshaled back to the input.
+func FuzzMicroTime(b []byte) int {
+	var unmarshalResult struct {
+		T metav1.MicroTime `json:"t"`
+	}
+	if err := yaml.Unmarshal(b, &unmarshalResult); err != nil {
+		return 0
+	}
+	marshalResult, err := yaml.Marshal(&unmarshalResult)
+	if err != nil {
+		panic(err)
```

Comment:
is `panic()` an accepted pattern here?
should fuzz tests e.g. print the error and `return 1` instead?

Reply:
Yeah, this is the suggested go-fuzz pattern (https://github.com/dvyukov/go-fuzz). This is also consistent with how the other fuzz targets tests/fuzz/json are written.

Reply:
@neolit123 basically if we trigger crashes to signal to the fuzzer that something is broken and needs looking at.